libbeat/outputs/elasticsearch/client.go

```diff
@@ -41,7 +40,7 @@ import (
 
 // Client is an elasticsearch client.
 type Client struct {
```

Comment:
I wonder if we can move some more fields to esclientleg.Connection. Ultimately we don't want to export `Client` anymore. And the exported `Client` symbol should become `type Client = esclientleg.Connection`.

Reply:
I managed to move some more fields to `esclientleg.Connection`. Now the only fields that remain are:

```
type Client struct {
	esclientleg.Connection

	index    outputs.IndexSelector
	pipeline *outil.Selector
	timeout  time.Duration

	observer outputs.Observer

	log *logp.Logger
}```

All of which, I believe, are specific to the ES output. WDYT?

Reply:
I was able to move `timeout` to `eslegclient.Connection` as well.